src/range/Range.ts

```diff
@@ -94,6 +94,22 @@ export default class Range {
     return r;
   }
 
+
+  concat(other: Range, size?: number[]) {
+    if (this.isAll || other.isNone) {
```

Comment:
does it make sense, if this or the other contain all ids, not to concatenate them?